### PR TITLE
subclasses of pet and unknown enum added

### PR DESCRIPTION
**Pull request by ayten-github**

### CODE REVIEW

File: src/main/java/az/edu/strangers/Species.java [hunk 1]
```diff
@@ -3,7 +3,8 @@
 public enum Species {
     DOG("Dog", 4, true, false),
     CAT("Cat", 4, true, false),
-    BIRD("Bird", 2, false, true);
+    BIRD("Bird", 2, false, true),
+    Unknown("Unknown", 0, false, false);
```
Comment (thapdigov) [inline src/main/java/az/edu/strangers/Species.java]: Only an empty constructor can be created for Unknown

File: src/main/java/az/edu/strangers/Dog.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    @Override
+    public void respond() {
+        System.out.printf("Haw-Haw,I am %s!/n", nickName);
+    }
+
+    @Override
+    public void foul() {
+        System.out.println("Digging holes");
+    }
```
Comment (Shammad-Kosayev) [inline src/main/java/az/edu/strangers/Dog.java:18]: Dog should implement `foul` method from `Foulable` interface.

File: src/main/java/az/edu/strangers/DomesticCat.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+    @Override
+    public void respond() {
+        System.out.printf("Meow,I am %s!/n", nickName);
+    }
+
+    @Override
+    public void foul() {
+        System.out.println("Screaming furniture");
+    }
```
Comment (Shammad-Kosayev) [inline src/main/java/az/edu/strangers/DomesticCat.java:18]: DomesticCat should implement `foul` method from `Foulable` interface.

File: src/main/java/az/edu/strangers/Fish.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package az.edu.strangers;
+
+public class Fish extends Pet {
+
+    public Fish(String nickName, Integer age, Integer trickLevel, String[] habits) {
+        super(Species.BIRD, nickName, age, trickLevel, habits);
```
Comment (Shammad-Kosayev) [inline src/main/java/az/edu/strangers/Fish.java]: I don't think fish is a "bird" species

File: src/main/java/az/edu/strangers/DomesticCat.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+public class DomesticCat extends Pet {
+
+    public DomesticCat(String nickName, Integer age, Integer trickLevel, String[] habits) {
+        super(Species.CAT, nickName, age, trickLevel, habits);
+    }
+
+    @Override
+    public void respond() {
+        System.out.printf("Meow,I am %s!/n", nickName);
```
Comment (Shammad-Kosayev) [inline src/main/java/az/edu/strangers/DomesticCat.java]: not '/n', it's '\n'

File: src/main/java/az/edu/strangers/Fish.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+    }
+
+    @Override
+    public void respond() {
+        System.out.printf("Hi,I am %s!/n", nickName);
```
Comment (Shammad-Kosayev) [inline src/main/java/az/edu/strangers/Fish.java]: not '/n', it's '\n'

File: src/main/java/az/edu/strangers/HappyFamilyApp.java [hunk 7]
```diff
@@ -4,27 +4,34 @@ public class HappyFamilyApp {
...
+//        schedule[1][0] = DayOfWeek.TUESDAY.name();
+//        schedule[1][1] = "Play football";
+//        Human child = new Human("Michael", "Karleone", 1999, 90, schedule, family);
+//
+//        family.addChild(child);
+//
+//        System.out.println(family);
+//
+//        child.greetPet();
+//        child.describePet();
```
Comment (Shammad-Kosayev) [inline src/main/java/az/edu/strangers/HappyFamilyApp.java]: Please remove redundant comments

File: src/main/java/az/edu/strangers/Pet.java [hunk 8]
```diff
@@ -28,18 +28,17 @@ public Pet(Species species, String nickName, Integer age, Integer trickLevel, St
...
     }
 
-    public void respond() {
-        System.out.printf("Hello, owner. I am  %s I miss you /n", nickName);
-    }
+    public abstract void respond();
 
     public void foul() {
         System.out.println("I need to cover it up");
     }
```
Comment (Shammad-Kosayev) [inline src/main/java/az/edu/strangers/Pet.java:37]: Extract this method to `Foulable` interface, please

File: src/main/java/az/edu/strangers/Fish.java (hunk 4 above)
Comment (Shammad-Kosayev) [inline src/main/java/az/edu/strangers/Fish.java]: Species.BIRD shouldn't be in super's constructor. It should be in the class's constructor

File: src/main/java/az/edu/strangers/RoboCat.java [hunk 9]
```diff
@@ -0,0 +1,14 @@
+package az.edu.strangers;
+
+public class RoboCat extends Pet {
+
+    public RoboCat(String nickName, Integer age, Integer trickLevel, String[] habits) {
+        super(Species.CAT, nickName, age, trickLevel, habits);
```
Comment (Shammad-Kosayev) [inline src/main/java/az/edu/strangers/RoboCat.java]: Species.CAT shouldn't be in super's constructor. It should be in the class's constructor

File: src/main/java/az/edu/strangers/RoboCat.java [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+    }
+
+    @Override
+    public void respond() {
+        System.out.printf("Beep Bop. I am %s!/n", nickName);
```
Comment (Shammad-Kosayev) [inline src/main/java/az/edu/strangers/RoboCat.java]: not '/n', it's '\n'

**Review by Shammad-Kosayev — CHANGES_REQUESTED** [review]
Please pay attention

**Review by Shammad-Kosayev — APPROVED** [review]
looks good to me

**Pull request merged**